cgo/const.go

```diff
@@ -96,6 +160,65 @@ func parseConstExpr(t *tokenizer, precedence int) (ast.Expr, *scanner.Error) {
 }
 
 func parseIdent(t *tokenizer) (ast.Expr, *scanner.Error) {
+	// If the identifier is one of the parameters of this function-like macro,
+	// use the parameter value.
+	if val, ok := t.params[t.curValue]; ok {
+		return val, nil
+	}
+
+	if t.f != nil {
+		// Check whether this identifier is actually a macro "call" with
+		// parameters. In that case, we should parse the parameters and pass it
+		// on to a new invocation of parseConst.
+		if t.peekToken == token.LPAREN {
+			if cursor, ok := t.f.names[t.curValue]; ok && t.f.isFunctionLikeMacro(cursor) {
+				t.Next()
```

Comment:
`Next()` called twice - is it intentional? If so, would be good to have a comment here

Reply:
Yes this is intentional. There is the "cur" (current) token and the "peek" (next) token. At this point we know the current token and the peek token, and we want to skip ahead so that the "current" token is the first unknown token. Hence two `Next()` calls.

Reply:
Thanks, makes sense :ok_hand: